sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -686,3 +686,35 @@ def test_lasso_lars_vs_R_implementation():
 
     assert_array_almost_equal(r2, skl_betas2, decimal=12)
     ###########################################################################
+
+
+def test_lasso_lars_copyX_behaviour1():
+    """
+    Test that user input regarding copyX is not being overridden (it was until
+    at least version 0.21)
+
+    Correct behaviour is not to create a copy.
+
+    """
+    lasso_lars = LassoLarsIC(copy_X=False, precompute=False)
+    X = np.random.normal(0, 1, (100, 5))
+    X_copy = X.copy()
+    y = X[:, 2]
+    lasso_lars.fit(X, y)
+    assert not np.array_equal(X, X_copy)
```

Comment:
You could do this with `assert lasso_lars.copy_X == np.array_equal (X, X_copy)`. You could then try True/False in one function with parameterize.

Reply:
clever. I hadn't used parametrize before, I have added it in the latest commit.